src/responses.ts

```diff
@@ -63,6 +64,54 @@ export function ResponseSuccessJson<T>(o: T): IResponseSuccessJson<T> {
   };
 }
 
+/**
+ * Interface for a successful response returning a xml object.
+ */
+export interface IResponseSuccessXml<T>
+  extends IResponse<"IResponseSuccessXml"> {
+  readonly value: T; // needed to discriminate from other T subtypes
+}
+
+/**
+ * Returns a successful xml response.
+ *
+ * @param o The object to return to the client
+ */
+export function ResponseSuccessXml<T>(o: T): IResponseSuccessXml<T> {
+  const kindlessObject = Object.assign(Object.assign({}, o), {
```

Comment:
mmh when does `T`  have the `kind` attribute? `kind` is used to the response types, not for their contained `value`

Reply:
removed